### DISCUSSION TITLE
Running Appium C# framework in Azure Devops - any examples

### DISCUSSION THREAD

**Discussion by Stanzarectah** [appium/appium#18997] [Q&A]
Hi,
I am currently using Appium V2.0 and using the .NET framework to implement the test cases.
All is working well on my local Windows machine and I am able to run the test against a specific version of my app using Android emulator.

I plan to automate these test cases once changes to the  mobile app has been merged to a main branch in Azure DevOps.

Does anyone have any examples to share ?

I would like to run the test cases against my local machine where the Android emulator is installed through the DevOps pipeline. The main challenge is figuring out what tasks needs to be created and how to do this using Powershell. I am no DevOps guru so this quite new to me

**Comment by KazuCocoa** [comment]
https://discuss.appium.io/ could help I guess. Some repositories in Appium server/clients have GitHub Actions/Azure pipeline so they may also help to see possible example. (they may not be C# though)

**Comment by hellowrld2986** [comment]
@Stanzarectah  I am also in the same boat, I am using Appium2 and I have to run the test cases parallelly in Azure Devops. Please let me know if you find a solution. 
I am new to .Net. I would appreciate if someone else helping me. Thanks!

**Reply by Blue101black** [reply]
Hey @Stanzarectah and @hellowrld2986 did you guys figure out how to get set up with Azure devops?

Did you create Azure Vm's to run the emulator?
